### PR TITLE
Social icons animation in footer & "email required" message bug fixed.

### PR DESCRIPTION
**Pull request by adi-exe**
## Issue that this pull request solves

 Closes: #828 Footer Email Input Validation Message Appears Unnecessarily When Input is Not Provided; Social media icons fails animation on hover IWOC

## Proposed changes

1. Footer Email Input Validation:
When the email input box in the footer is active, the validation message indicating "email required" was appearing even when no input is provided. This has been fixed now, it doesn't show this message unnecessarily.

2. Social Media Icons Animation:
Previously on hovering over the social media icons in the footer, instagram , twitter , youtube failed to display the expected animation. I have added the particular animation to them.

3. Placeholder of email input:
The placeholder remained the same dark colored even on focus if the input .
I have added a light grey color to make is more pleasing to user.

## Types of changes

_Put an `x` in the boxes that apply_

- [ x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Screenshot (124)
Screenshot (125)
Screenshot (126)

## Other information

Any other information that is important to this pull request

### CODE REVIEW

**Review by thevirengarg — CHANGES_REQUESTED** [review]
The message "Email is required" should be visible if the email input field is focused but is empty and as soon as user starts typing the email it should clear the error message. If it's not selected it should display "Email ID".

**Comment by adi-exe** [comment]
so basically we have to remove " * Please enter valid Email " message ,
cause as soon as the user starts writing it shows this message.

File: frontend/src/components/Footer/Footer.jsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const Footer = (props) => {
       isValid = false;
     }
     if (!email.trim()) {
-      emailErr.emailRequired = "* Email is required";
```
Comment (Kajol-Kumari) [inline frontend/src/components/Footer/Footer.jsx:26]: retain this. This is an expected error message

File: frontend/src/components/Footer/footer.module.scss [hunk 2]
```diff
@@ -207,16 +212,26 @@
   color: #dd2a7b;
 }
 
-.fa-envelope:hover,
-.fa-envelope-own:hover {
-  color: #c71610;
+.fa-instagram:hover,
```
Comment (Kajol-Kumari) [inline frontend/src/components/Footer/footer.module.scss:215]: any specific reason of modfying the css? Why do we need to change these hover effects?

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]

**Comment by adi-exe** [comment]
@Kajol-Kumari @thevirengarg  pls reply to these changes, IWOC is ending soon.

File: frontend/src/components/Footer/footer.module.scss (hunk 2 above)
Comment (Kajol-Kumari) [reply frontend/src/components/Footer/footer.module.scss:216]: Its a favicon icon, so the icons u see on the footer for instagram is fa-instagram

**Review by Kajol-Kumari — APPROVED** [review]

**Pull request merged**